### PR TITLE
 Add module for bruteforcing authentication on MQTT endpoints

### PR DESCRIPTION
**Pull request by jhart-r7**
This adds a module for brute forcing authentication on MQTT endpoints.

Note that this PR appears to include new mixins and new library code, however that is due to the branching strategy I used.  The review of that code is in #9329 and once that lands this PR will be updated to reflect reality.

## Verification

Follow steps in provided documentation.

### CODE REVIEW

File: lib/metasploit/framework/login_scanner/mqtt.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+              read_timeout: read_timeout,
+              client_id: client_id
+            }
+            client = Rex::Proto::MQTT::Client.new(sock, client_opts)
+            connect_res = client.connect
+            client.disconnect
+
+            if connect_res.return_code == 0
+              status = Metasploit::Model::Login::Status::SUCCESSFUL
+              proof = "Succesful Connack"
```
Comment (busterb) [inline lib/metasploit/framework/login_scanner/mqtt.rb]: what's a Connack ? At first I thought it was a misspelling of 'Connection'

Comment (wvu-r7) [reply lib/metasploit/framework/login_scanner/mqtt.rb]: And maybe s/Succesful/Successful/, too.

Comment (busterb) [reply lib/metasploit/framework/login_scanner/mqtt.rb]: answering myself: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718033

**Comment by busterb** [comment]
Added a couple of minor changes:

ae17943d4c - fix documentation preformat blocks (2 minutes ago) <Brent Cook>
6f1196d30c - clarify what's happening when there is a connection failure (2 minutes ago) <Brent Cook>

Thanks @jhart-r7

**Comment by busterb** [comment]
## Release Notes

This adds a module for brute forcing authentication on MQTT endpoints.

**Pull request merged**